lib/rubocop/cop/lint/non_deterministic_require_order.rb

```diff
@@ -61,45 +63,43 @@ def on_block(node)
           loop_variable(node.arguments) do |var_name|
             return unless var_is_required?(node.body, var_name)
 
-            add_offense(node.send_node)
+            add_offense(node.send_node) do |corrector|
+              autocorrect(corrector, node.send_node)
+            end
           end
         end
 
         def on_block_pass(node)
           return unless method_require?(node)
           return unless unsorted_dir_pass?(node.parent)
 
-          add_offense(node.parent)
+          add_offense(node.parent) do |corrector|
+            autocorrect(corrector, node.parent)
```

Comment:
One of the goal of `Base` is to avoid calling a single entry point like `autocorrect` here and redoing an `if` there. Can't we call `correct_block_pass` directly here?

Reply:
This did not have to be a single entry point. I updated it.